### PR TITLE
Fix all design errors + making QR extra neat

### PR DESCRIPTION
**Pull request by clenrock**
- Relevant Issues: #
- Related Pull Requests: -

* * *

## What
This Pull Request fixes all the design errors which can be seen at the front-end of our app.

## Why
This Pull Request is needed because having a perfect design makes the app look much more professional. 

## How
This feature can be viewed/tested within the project by opening the app and clicking through it and checking if something LOOKS weird. Also note the new QR activity ordering. 

## Notes
@ericcornelissen Sorry for changing the icon dimensions agian, but some icons were 2x too large.

### CODE REVIEW

File: app/src/main/res/layout/activity_qrexchange.xml [hunk 1]
```diff
@@ -14,41 +14,80 @@
...
+                    tools:ignore="ContentDescription" /> <!-- Empty button doesn't need a description -->
+
+            </LinearLayout>
+
+        </android.support.design.widget.AppBarLayout>
+
     </android.support.design.widget.AppBarLayout>
 
+    <FrameLayout
+        android:layout_width="400dp"
```
Comment (jverbraeken) [inline app/src/main/res/layout/activity_qrexchange.xml]: Starting from here, the next 40 lines contain a lot of hardcoded integer constants. Are you sure that these shouldn't be in dimens.xml?

File: app/src/main/java/com/nervousfish/nervousfish/activities/QRExchangeKeyActivity.java [hunk 2]
```diff
@@ -69,6 +69,8 @@ protected void onCreate(final Bundle savedInstanceState) {
         final KeyPair pair = keyGenerator.generateRSAKeyPair("test");
         publicKey = pair.getPublicKey();
 
+        showQRCode();
+
```
Comment (ericcornelissen) [inline app/src/main/java/com/nervousfish/nervousfish/activities/QRExchangeKeyActivity.java]: Could you remove the additional enter 😇 

Also, this PR seems like the perfect oppertunity to resolve the `//TODO:` [a few lines up](https://github.com/ericcornelissen/NervousFish/pull/224/files#diff-3ee102e75053006952e1d321d14f640dR67) 😄

File: app/src/main/java/com/nervousfish/nervousfish/activities/QRExchangeKeyActivity.java [hunk 3]
```diff
@@ -86,10 +88,7 @@ public void onBackButtonClick(final View view) {
      */
     public void onShowQRButtonClick(final View view) {
         LOGGER.info("Started generating QR code");
-        final String space = " ";
-        final Bitmap qrCode = QRGenerator.encode(publicKey.getType() + space + publicKey.getName()
-                + space + publicKey.getKey());
-        showQRCode(qrCode);
+//        showQRCode(qrCode);
```
Comment (ericcornelissen) [inline app/src/main/java/com/nervousfish/nervousfish/activities/QRExchangeKeyActivity.java]: Can you remove commented code. And at that point, can you remove this method? 😄

File: app/src/main/java/com/nervousfish/nervousfish/activities/QRExchangeKeyActivity.java [hunk 4]
```diff
@@ -148,25 +147,27 @@ private void addNewContact(final String qrMessage) {
...
-    private void showQRCode(final Bitmap qrCode) {
+    private void showQRCode() {
 
-        final LayoutInflater li = LayoutInflater.from(this);
-        final View myView = li.inflate(R.layout.qrcode, null);
+        final String space = " ";
+        final Bitmap qrCode = QRGenerator.encode(publicKey.getType() + space + publicKey.getName()
+                + space + publicKey.getKey());
+//        final LayoutInflater li = LayoutInflater.from(this);
+//        final View myView = li.inflate(R.layout.qrcode, null);
```
Comment (ericcornelissen) [inline app/src/main/java/com/nervousfish/nervousfish/activities/QRExchangeKeyActivity.java]: Can you remove commented code 😇

File: app/src/main/java/com/nervousfish/nervousfish/activities/QRExchangeKeyActivity.java [hunk 5]
```diff
@@ -148,25 +147,27 @@ private void addNewContact(final String qrMessage) {
...
-        lastDialog = builder.create();
-        lastDialog.show();
+        imageView.setMaxWidth(30);
+//        final AlertDialog.Builder builder = new AlertDialog.Builder(this)
+//                .setView(imageView)
+//                .setPositiveButton(getString(R.string.done), new QRCloser());
+//
+//        ((ViewGroup) imageView.getParent()).removeView(imageView);
+//        lastDialog = builder.create();
+//        lastDialog.show();
```
Comment (ericcornelissen) [inline app/src/main/java/com/nervousfish/nervousfish/activities/QRExchangeKeyActivity.java]: Can you remove commented code 😇 

Also can you removed the unused code on [line 174](https://github.com/ericcornelissen/NervousFish/pull/224/files#diff-3ee102e75053006952e1d321d14f640dR174)

File: app/src/main/res/layout/activity_bluetooth_connection.xml [hunk 6]
```diff
@@ -12,32 +13,23 @@
...
-            android:layout_height="match_parent"
-            android:orientation="horizontal">
+            android:layout_height="?attr/actionBarSize"
+            android:layout_weight="1"
+            android:background="@android:color/holo_blue_dark"
+            app:popupTheme="@style/AppTheme.PopupOverlay">
 
             <TextView
-                android:id="@+id/toolbar_title"
+                android:id="@+id/toolbar_title_main"
```
Comment (ericcornelissen) [inline app/src/main/res/layout/activity_bluetooth_connection.xml]: Is this `toolbar_title_main`? Remember that the android:id namespace is global :wink:

**Comment by ericcornelissen** [comment]
> Sorry for changing the icon dimensions agian, but some icons were 2x too large.

No problem, if Android Lint doesn't complain, I don't complain 😄

File: app/src/main/res/layout/activity_qrexchange.xml (hunk 1 above)
Comment (clenrock) [inline app/src/main/res/layout/activity_qrexchange.xml]: Thanks for reminding me. I always try to get a nice design by quickly changing the parameter, and in the end put them in dimens.xml. I forgot to do that now

**Comment by clenrock** [comment]
@TheBonheurs Sure, but these were all very small errors in the design which takes more time to document than to fix

**Comment by ericcornelissen** [comment]
@clenrock still, it may be obvious to you because you made the changes, but not everyone notices everything you changed 😉

File: app/src/main/java/com/nervousfish/nervousfish/activities/QRExchangeKeyActivity.java [hunk 7]
```diff
@@ -113,113 +74,16 @@ public void onScanButtonClick(final View view) {
...
-
     /**
-     * Shows the QR Code in an alert dialog popup screen
-     *
-     * @param qrCode The QR code to be shown.
+     * Shows the QR Code in the activity.
      */
     @SuppressLint("InflateParams")
-    private void showQRCode(final Bitmap qrCode) {
+    private void showQRCode() {
```
Comment (ericcornelissen) [inline app/src/main/java/com/nervousfish/nervousfish/activities/QRExchangeKeyActivity.java]: Can you rename this to `drawQRCode` 😇

**Pull request merged**